src/bundles/arcade_2d/phaserScene.ts

```diff
@@ -0,0 +1,358 @@
+import Phaser from 'phaser';
+import {
+  CircleGameObject,
+  GameObject,
+  type InteractableGameObject,
+  RectangleGameObject,
+  ShapeGameObject,
+  SpriteGameObject,
+  TextGameObject,
+  TriangleGameObject,
+} from './gameobject';
+import {
+  CONFIG,
+} from './functions';
+import {
+  type TransformProps,
+  type PositionXY,
+} from './types';
+import { AudioClip } from './audio';
+
+type PhaserGameObject = Phaser.GameObjects.Sprite | Phaser.GameObjects.Text | Phaser.GameObjects.Shape;
+
+// Game state information, that changes every frame.
+export const gameState = {
+  // Stores the debug information, which is reset every iteration of the update loop.
+  debugLogArray: Array.of() as Array<string>,
+  // The current in-game time and frame count.
+  gameTime: 0,
+  loopCount: 0,
+  // Store keys that are down in the Phaser Scene
+  // By default, this is empty, unless a key is down
+  inputKeysDown: new Set<string>(),
+  pointerProps: {
+    // the current (mouse) pointer position in the canvas
+    pointerPosition: [0, 0] as PositionXY,
+    // true if (left mouse button) pointer down, false otherwise
+    pointerPrimaryDown: false,
+    pointerSecondaryDown: false,
+    // Stores the IDs of the GameObjects that the pointer is over
+    pointerOverGameObjectsId: new Set<number>(),
+  },
+};
+
+// The game state which the user can modify, through their update function.
+const userGameStateArray: Array<any> = Array.of();
+
+/**
+ * The Phaser scene that parses the GameObjects and update loop created by the user,
+ * into Phaser GameObjects, and Phaser updates.
+ */
+export class PhaserScene extends Phaser.Scene {
+  constructor() {
+    super('PhaserScene');
+  }
+  private sourceGameObjects;
+  private phaserGameObjects = [] as (PhaserGameObject)[];

```

Comment:
I'm not sure why the type cast here is necessary. Did you mean:

```suggestion
  private phaserGameObjects: Array<PhaserGameObject> = [];
```